### PR TITLE
Adds `array::intersects()`

### PR DESCRIPTION
**Pull request by atombender**

### CODE REVIEW

File: spec/11-functions.md [hunk 1]
```diff
@@ -552,6 +552,36 @@ array_unique_validate(args):
...
+  - Return {null}.
+- Let {secondNode} be the first element of {args}.
+- Let {second} be the result of {Evaluate(secondNode, scope)}.
+- If {second} is not an array:
+  - Return {null}.
+- For each element in {first}:
+  - Let {a} be the element
+  - For each element in {second}:
+    - Let {b} be the element.
+    - Set {equal} be the result of `Equal(a, b)`
```
Comment (sgulseth) [inline spec/11-functions.md:575]:
```suggestion
    - Set {equal} to be the result of `Equal(a, b)`
```

**Review by sgulseth — APPROVED** [review]

File: spec/11-functions.md [hunk 2]
```diff
@@ -552,6 +552,36 @@ array_unique_validate(args):
...
+The `intersects` function compares two arrays, returning `true` if they have any elements in common.
+
+Only values that can be compared for [equality](#sec-Equality) are considered when determining whether there are common values.
+
+array_intersects(args, scope):
+
+- Let {firstNode} be the first element of {args}.
+- Let {first} be the result of {Evaluate(firstNode, scope)}.
+- If {first} is not an array:
+  - Return {null}.
```
Comment (mwain) [inline spec/11-functions.md:566]: Why return null and not false?

Comment (sgulseth) [reply spec/11-functions.md:566]: I was thinking the same, but it aligns with other methods. Ie `string::startsWith` returns `null` if arg is not a string

Comment (atombender) [reply spec/11-functions.md:566]: Yes, see explanation [here](https://github.com/sanity-io/GROQ/blob/c62f4f39cc48e7c530a2ffe740822a0b1e1184b3/spec/04-data-types.md#L59).

**Review by judofyr — APPROVED** [review]

**Pull request merged**